src/commands/createFunctionApp/stacks/FunctionAppEOLWarningStep.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { AzureWizardPromptStep, DialogResponses, openUrl } from "@microsoft/vscode-azext-utils";
+import { MessageItem } from "vscode";
+import { funcVersionLink } from "../../../FuncVersion";
+import { localize } from "../../../localize";
+import { getWorkspaceSetting, updateGlobalSetting } from "../../../vsCodeConfig/settings";
+import { IFunctionAppWizardContext } from "../IFunctionAppWizardContext";
+
+export class FunctionAppEOLWarningStep extends AzureWizardPromptStep<IFunctionAppWizardContext> {
+    public async prompt(context: IFunctionAppWizardContext): Promise<void> {
+        const settingKey: string = 'endOfLifeWarning';
+        if (getWorkspaceSetting<boolean>(settingKey)) {
+            const message = localize('endOfLife', "The chosen runtime stack has an end of support deadline coming up." +
+                "After the deadline, function apps can be created and deployed, and existing apps continue to run." +
+                "However, your apps won't be eligible for new features, security patches, performance optimizations, and support until you upgrade them");
+            const result: MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.learnMore, DialogResponses.dontWarnAgain);
+            if (result === DialogResponses.learnMore) {
+                await openUrl(funcVersionLink);
```

Comment:
Clicking learn more opens the link as expected, but it also advances to the next step. I think the preferred behavior would be to have the modal stay up after clicking learn more.